src/main/java/org/openpnp/scripting/Scripting.java

```diff
@@ -126,4 +161,24 @@ public void on(String event, Map<String, Object> globals) throws Exception {
             }
         }
     }
+
+    public void clearScriptingEnginesCache() {
+        if (engineCache.size() == 0) {
+            Logger.info("No scripting engines cached, nothing to do");
+        }
+
+        List<String> clearedEngines = new ArrayList<String>();
+        for (Map.Entry<String, ScriptEngine> entry : engineCache.entrySet()) {
+            String engineInfo = "[" + entry.getKey() + ", " + entry.getValue() + "]";
+            clearedEngines.add(engineInfo);
+        }
+        engineCache.clear();
+        Logger.info("Cleared " + clearedEngines.size() + " scripting engines from cache: "
+                + String.join(", ", clearedEngines));
+    }
+
+    // To be used to disable/enable the cache clear menu button
+    // public int getCachedScriptingEnginesCount() {
+    //     return engineCache.size();
+    // }
```

Comment:
This can be removed or used when the open questions are addressed.

Reply:
Sorry, I don't understand what you mean by "open questions", i.e. how it is related to this piece of code?

Reply:
resolved in https://github.com/openpnp/openpnp/pull/1475#issuecomment-1229542829